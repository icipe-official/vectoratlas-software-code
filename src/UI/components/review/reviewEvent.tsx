import { ReviewEvent } from './ReviewForm';
import FileUploadIcon from '@mui/icons-material/FileUpload';
import RemoveRedEyeIcon from '@mui/icons-material/RemoveRedEye';
import DoneOutlineIcon from '@mui/icons-material/DoneOutline';
import { ListItem, ListItemIcon, ListItemText, useTheme } from '@mui/material';

export function ReviewEventItem({ event }: { event: ReviewEvent }) {
  const theme = useTheme();
  const eventString = `${event.type} by ${event.performedBy} on ${event.performedAt}.`;
  return (
    <ListItem>
      <ListItemIcon
        sx={{
          minWidth: 0,
          mr: 'auto',
          justifyContent: 'center',
        }}
      >
        {event.type === 'Uploaded' ? (
          <FileUploadIcon
            sx={{ color: theme.palette.secondary.main, margin: '5px' }}
          />
        ) : event.type === 'Reviewed' ? (
          <RemoveRedEyeIcon sx={{ color: 'blue', margin: '5px' }} />
        ) : (
          <DoneOutlineIcon
            sx={{ color: theme.palette.primary.main, margin: '5px' }}
          />
        )}
      </ListItemIcon>
      <ListItemText primary={eventString} />
    </ListItem>
  );
}
