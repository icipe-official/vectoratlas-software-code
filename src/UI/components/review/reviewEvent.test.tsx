import { render } from '../../test_config/render';
import { ReviewEventItem } from './reviewEvent';

jest.mock('@mui/icons-material/FileUpload',
  () =>
    function FileUploadsMock() {
      return <div>FileUpload mock</div>;
    }
);
jest.mock('@mui/icons-material/RemoveRedEye',
  () =>
    function RemoveRedEyeMock() {
      return <div>RemoveRedEye mock</div>;
    }
);
jest.mock('@mui/icons-material/DoneOutline',
  () =>
    function DoneOutlineMock() {
      return <div>DoneOutline mock</div>;
    }
);

describe('ReviewEventItem', () => {
  it('renders for In review event', () => {
    const event = {
      type: 'Reviewed',
      performedBy: 'user123',
      performedAt: '2023-02-02T10:35:50.061Z'
    };
    const { wrapper } = render(<ReviewEventItem event={event} />)
    expect(wrapper.container).toMatchSnapshot();
  });

  it('renders for Uploaded event', () => {
    const event = {
      type: 'Uploaded',
      performedBy: 'user123',
      performedAt: '2023-02-02T10:35:50.061Z'
    };
    const { wrapper } = render(<ReviewEventItem event={event} />)
    expect(wrapper.container).toMatchSnapshot();
  });

  it('renders for Approved event', () => {
    const event = {
      type: 'Approved',
      performedBy: 'user123',
      performedAt: '2023-02-02T10:35:50.061Z'
    };
    const { wrapper } = render(<ReviewEventItem event={event} />)
    expect(wrapper.container).toMatchSnapshot();
  });
});
