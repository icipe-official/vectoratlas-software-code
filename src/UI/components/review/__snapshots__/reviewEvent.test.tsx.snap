// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewEventItem renders for Approved event 1`] = `
<div>
  <li
    class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
  >
    <div
      class="MuiListItemIcon-root css-1ddsiuw-MuiListItemIcon-root"
    >
      <div>
        DoneOutline mock
      </div>
    </div>
    <div
      class="MuiListItemText-root css-tlelie-MuiListItemText-root"
    >
      <span
        class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
      >
        Approved by user123 on 2023-02-02T10:35:50.061Z.
      </span>
    </div>
  </li>
</div>
`;

exports[`ReviewEventItem renders for In review event 1`] = `
<div>
  <li
    class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
  >
    <div
      class="MuiListItemIcon-root css-1ddsiuw-MuiListItemIcon-root"
    >
      <div>
        RemoveRedEye mock
      </div>
    </div>
    <div
      class="MuiListItemText-root css-tlelie-MuiListItemText-root"
    >
      <span
        class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
      >
        Reviewed by user123 on 2023-02-02T10:35:50.061Z.
      </span>
    </div>
  </li>
</div>
`;

exports[`ReviewEventItem renders for Uploaded event 1`] = `
<div>
  <li
    class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
  >
    <div
      class="MuiListItemIcon-root css-1ddsiuw-MuiListItemIcon-root"
    >
      <div>
        FileUpload mock
      </div>
    </div>
    <div
      class="MuiListItemText-root css-tlelie-MuiListItemText-root"
    >
      <span
        class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
      >
        Uploaded by user123 on 2023-02-02T10:35:50.061Z.
      </span>
    </div>
  </li>
</div>
`;
